controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java

```diff
@@ -0,0 +1,153 @@
+
+    public PasswordAuthHandler() {
+        userMap = new ConcurrentHashMap<>();
+    }
+
+    private void loadPasswordFile(String userPasswordFile) {
+        try (FileReader reader = new FileReader(userPasswordFile);
+             BufferedReader lineReader = new BufferedReader(reader)) {
+            String line;
+            while ( !Strings.isNullOrEmpty(line = lineReader.readLine())) {
+                if (line.startsWith("#")) {
+                    continue;
+                }
+                String[] userFields = line.split(":");
+                if (userFields.length >= 2) {
+                    String acls;
+                    if (userFields.length == 2) {
+                        acls = "";
+                    } else {
+                        acls = userFields[2];
+                    }
+                    userMap.put(userFields[0], new PravegaACls(userFields[1], getAcls(acls)));
+                }
+            }
+        } catch (IOException e) {
+            throw new CompletionException(e);
+        }
+    }
+
+    @Override
+    public String getHandlerName() {
+        return DEFAULT_NAME;
+    }
+
+    @Override
+    public boolean authenticate(Map<String, String> headers) {
+        String userName = headers.get("username");
+        String password = headers.get("password");
+        Preconditions.checkArgument(userName != null, "Username not found in header");
+        Preconditions.checkArgument(password != null, "Password not found in header");
+
+        StrongPasswordProcessor encryptor = new StrongPasswordProcessor();
+        try {
+            return userMap.containsKey(userName) && encryptor.checkPassword(password, userMap.get(userName).encryptedPassword);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            log.warn("Excpetion during password authentication", e);
+            return false;
+        }
+    }
+
+    @Override
+    public PravegaAccessControlEnum authorize(String resource, Map<String, String> headers) {
+        String userName = headers.get("username");
+        if (Strings.isNullOrEmpty(userName) || !userMap.containsKey(userName)) {
+            throw new CompletionException(new PravegaAuthenticationException(userName));
+        }
+        return authorizeForUser(userMap.get(userName), resource);
+
+    }
+
+    @Override
+    public void initialize(Object serverConfig) {
+        loadPasswordFile(((GRPCServerConfig) serverConfig).getUserPasswordFile());
+    }
+
+    private PravegaAccessControlEnum authorizeForUser(PravegaACls pravegaACls, String resource) {
+        PravegaAccessControlEnum retVal = PravegaAccessControlEnum.NONE;
+
+        /**
+         *  `*` Means a wildcard.
+         *  If It is a direct match, return the ACLs.
+         *  If it is a partial match, the target has to end with a `/`
+         */
+        for (PravegaAcl acl : pravegaACls.acls) {
+            if (acl.resource.equals(resource)) {
+                return acl.acl;
```

Comment:
I'm confused by this conditional logic. There are several branches but they all return the same thing.

Reply:
Simplified the logic.